### PR TITLE
Fixed PostMessage error bug.  fixed #294

### LINKED ISSUE #294: PostMessage doesn't return error if error occurs
I requested to post message to slack with a incorrect channel.
expected error (`channel_not_found`) but returned nil.

code:
```go
func PostToSlack(channel string, msg string, params ...slack.PostMessageParameters) (err error) {
	token := (MY SLACK TOKEN)

	api := slack.New(token)
        api.SetDebug(true) // for debug
	p := slack.PostMessageParameters{}
	if len(params) > 0 {
		p = params[0]
	}
	// _, _, err = api.PostMessage(channel, msg, p)
	a, b, err = api.PostMessage(channel, msg, p) // for debug
        spew.Dump(a, b, err) // dump message for debug -- github.com/davecgh/go-spew/spew
	if err != nil {
		return err
	}
	return
}
```

test:
```go
func TestPostSlackWithInvalidChannel(t *testing.T) {
	err := PostToSlack("#foobar", "test")
	assert.EqualError(t, err, "channel_not_found")
}
```

test result:
```
=== RUN   TestPostSlackWithInvalidChannel
nlopes/slack2018/04/05 16:35:28 <autogenerated>:1: parseResponseBody: {"ok":false,"error":"channel_not_found"}
(string) ""
(string) ""
(interface {}) <nil>
--- FAIL: TestPostSlackWithInvalidChannel (0.22s)
	notify_test.go:37: 
			Error Trace:	notify_test.go:37
			Error:      	An error is expected but got nil.
			Test:       	TestPostSlackWithInvalidChannel
FAIL
```

### PR DESCRIPTION
**Pull request by asiyani**
Added condition to check `SlackResponse` for any Error.
[info](https://api.slack.com/web)
```
{
    "ok": true,
    "stuff": "This is good"
}
```
```
{
    "ok": false,
    "error": "something_bad"
}
```
if `SlackResponse.Ok` is false `SendMessageContext` will return an Error.

If any issue please let me know.

Thank you.

### CODE REVIEW

**Comment by james-lawrence** [comment]
yikes... let me take a look at how that happened.

File: chat.go [hunk 1]
```diff
@@ -157,6 +158,11 @@ func (api *Client) SendMessageContext(ctx context.Context, channelID string, opt
 		return "", "", "", err
 	}
 
+	//Check slack response for error
+	if response.SlackResponse.Ok == false {
+		return "", "", "", fmt.Errorf("slack API error: %s", response.SlackResponse.Error)
```
Comment (james-lawrence) [inline chat.go:163]: two things:
1. lets return the error string directly. don't prepend 'slack API error:' to it.
2. lets make this a function on SlackResponse.

```golang
type SlackResponse struct {
	Ok    bool   `json:"ok"`
	Error string `json:"error"`
}

func (t SlackResponse) Err() error {
  if t.Ok {
    return nil
  }

  return errors.New(t.Error)
}
```

**Review by james-lawrence — COMMENTED** [review]
Will accept as is, but mind adding the test case from the issue.

**Comment by james-lawrence** [comment]
LGTM, once the error handling is moved to the SlackResponse struct.

**Comment by james-lawrence** [comment]
closing in favor of #307, which includes this fix and some error handling cleanup and a test.

**Comment by james-lawrence** [comment]
thanks for the PR pointing out where to apply the fix.

**Comment by asiyani** [comment]
I am sorry for not getting back to you on time.

**Pull request closed without merging**